src/frontend/src/session.rs

```diff
@@ -102,6 +102,10 @@ pub struct FrontendEnv {
     source_metrics: Arc<SourceMetrics>,
 
     batch_config: BatchConfig,
+
+    pub distributed_query_limit: Option<u64>,
+
+    pub conrruent_distributed_query_count: Arc<AtomicU64>,
```

Comment:
I think it's better to move this to `QueryManager`, since all distributed query will go to it. Please also add an metric called `batch_running_distributed_query` in frontend.